lib/galaxy/util/checkers.py

```diff
@@ -16,13 +16,17 @@
 
 HTML_CHECK_LINES = 100
 CHUNK_SIZE = 2 ** 15  # 32Kb
-HTML_REGEXPS = (
-    re.compile(r"<A\s+[^>]*HREF[^>]+>", re.I),
-    re.compile(r"<IFRAME[^>]*>", re.I),
-    re.compile(r"<FRAMESET[^>]*>", re.I),
-    re.compile(r"<META[\W][^>]*>", re.I),
-    re.compile(r"<SCRIPT[^>]*>", re.I),
-)
+HTML_REGEXPS = {
+    "link": re.compile(r"<A\s+[^>]*HREF[^>]+>", re.I),
+    "iframe": re.compile(r"<IFRAME[^>]*>", re.I),
+    "frameset": re.compile(r"<FRAMESET[^>]*>", re.I),
+    "meta": re.compile(r"<META[\W][^>]*>", re.I),
+    "script": re.compile(r"<SCRIPT[^>]*>", re.I),
+}
+
+HTML_ALLOWLIST = {
+    "meta": re.compile(r'<meta name="qrichtext" content="1"\s*/>', re.I),
```

Comment:
If I google for "meta qrichtext" I see mainly stuff related to Qt QTextEdit, so maybe that's a red herring?

Reply:
The underlying problem is really that any HTML can be embedded. From all test files that I had, this at least made it possible to load 90% of all files. I hope someone else has an idea here, how we can solve this, or if you should simply support smbl.gz files only.